cache/clock_cache.h

```diff
@@ -295,28 +567,38 @@ class ClockHandleTable {
  private:
   friend class ClockCacheShard;
 
-  int FindVisibleElement(const Slice& key, int& probe, int displacement);
+  int FindElement(const Slice& key, uint32_t hash, uint32_t& probe);
 
-  int FindAvailableSlot(const Slice& key, int& probe, int displacement);
+  int FindAvailableSlot(const Slice& key, uint32_t& probe);
 
-  int FindVisibleElementOrAvailableSlot(const Slice& key, int& probe,
-                                        int displacement);
+  int FindElementOrAvailableSlot(const Slice& key, uint32_t hash,
+                                 uint32_t& probe);
 
   // Returns the index of the first slot probed (hashing with
-  // the given key) with a handle e such that cond(e) is true.
-  // Otherwise, if no match is found, returns -1.
-  // For every handle e probed except the final slot, updates
-  // e->displacements += displacement.
-  // The argument probe is modified such that consecutive calls
-  // to FindSlot continue probing right after where the previous
-  // call left.
-  int FindSlot(const Slice& key, std::function<bool(ClockHandle*)> cond,
-               int& probe, int displacement);
+  // the given key) with a handle e such that match(e) is true.
+  // At every step, the function first tests whether match(e) holds.
+  // If it's false, it evaluates abort(e) to decide whether the
+  // search should be aborted, and in the affirmative returns -1.
+  // For every handle e probed except the last one, the function runs
+  // update(e). We say a probe to a handle e is aborting if match(e) is
+  // false and abort(e) is true. The argument probe is one more than the
+  // last non-aborting probe during the call. This is so that that the
+  // variable can be used to keep track of progress across consecutive
+  // calls to FindSlot.
+  inline int FindSlot(const Slice& key, std::function<bool(ClockHandle*)> match,
```

Comment:
Could you please add a short description about the probing algorithm (e.g. linear probing, quadratic probing , etc)? You can add this in the function body.